drivers/stepper/adi/Kconfig

```diff
@@ -0,0 +1,50 @@
+# SPDX-FileCopyrightText: Copyright (c) 2024 Carl Zeiss Meditec AG
+# SPDX-License-Identifier: Apache-2.0
+
+menuconfig STEPPER_ADI_TMC
+	bool "Trinamic Stepper Controller"
+	depends on STEPPER
+	default y
+	help
+	  Enable trinamic stepper controller
+
+config STEPPER_ADI_TMC_RAMP_GEN
+	bool "Use Trinamic Stepper Controller with Ramp Generator"
+	depends on STEPPER_ADI_TMC
+	default y
+	help
+	  If y, the Trinamic Stepper Controller uses Ramp Generator
+
+config STEPPER_ADI_TMC_STALLGUARD
+	bool "Use Trinamic Stepper Controller with StallGuard"
+	depends on STEPPER_ADI_TMC
+	default y
+	help
+	  If y, the Trinamic Stepper Controller uses StallGuard
+
+if STEPPER_ADI_TMC_STALLGUARD
+
+config STEPPER_ADI_TMC_STALLGUARD_ACTIVATION_PAUSE_IN_MSEC
+	int "StallGuard Activation Pause in ms"
+	depends on STEPPER_ADI_TMC_STALLGUARD
+	default 100
+	help
+	  StallGuard Activation Pause in ms.
+	  StallGuard should not be enabled during motor spin-up.
+	  Wait for the motor to reach a stable speed before enabling StallGuard.
+
+endif # STEPPER_ADI_TMC_STALLGUARD
```

Comment:
would it be a good idea to use 
`DT_STRING_UNQUOTED(node_id, prop)
`
then in device tree you could use something like 
```
stepper: stepper@0{
     use-stallguard;
     stallguard-delay = "K_MSEC(100)";
}
```

Reply:
I was not even sure what to say when I say `stallguard-delay = "K_MSEC(100)";` on my inbox. Please keep in mind DT needs to avoid any sort of zephyrisms, hacks, etc. Just use `stallguard-delay-ms = <100>`.

Reply:
The same could be applied for the RAMP_GENERATOR part as well, where you could activate ramp-gen from the dt bindings. This entire thing could be removed from kconfig alltogether then i suppose

Reply:
The "same" can't be applied to any property. Please read the DT-spec or bindings docs. `"K_MSEC(100)";` in DT just doesn't make any sense.

Reply:
To summarize a bit, things that describe hardware, can be in the devicetree. Things like interrupt routing, a flag indicating whether the specific hardware supports some feature, how many of some thing that are present in a specific instance of hardware, like number of RX mailboxes of a specific CAN instance.

Things that do not go in devicetree are anything that is specific to a driver or subsystem, like the size of some buffer, or lazy delay used in a driver.

There are exceptions to everything, but generally, anything in the devicetree should be usable by any driver, even a non-zephyr driver.

Reply:
> would it be a good idea to use
> `DT_STRING_UNQUOTED(node_id, prop) `

Yeah, no don't do that :-)

`stallguard-delay-ms = <xxx>;`